poc/ws_client.py

```diff
@@ -0,0 +1,170 @@
+"""
+Client Websocket Management
+
+TODO: Add Exception Handling for websocket disconnect
+"""
+
+import asyncio
+import threading
+from typing import Awaitable
+
+import asgiref.sync
+from gui import ChatClientGUI
+from websockets import connect
+
+
+class ExampleClient:
+    """Example Client that illustrates how to use the ClientConnection API"""
+
+    @asgiref.sync.sync_to_async
+    def producer(self):
+        """
+        An example producer
+
+         - Producers should be awaitable/async
+         - Producers are functions that return data after calculations/processing has taken place
+         - They should return data at intervals
+        """
+        # Reading stdin is a blocking call
+        txt = input("")
+        return txt
+
+    async def consumer(self, msg):
+        """
+        An example consumer
+
+         - Consumers should be awaitable/async
+         - Consumers handle/process received data
+        Args:
+            msg (Any): Any type of data. Data is transmitted over websockets as bytes or strings
+        """
+        # Do something with the received msg
+        print(msg, end="\n>>>")
+
+
+class ClientConnection:
+    """Websocket Client Connection"""
+
+    def __init__(
+        self, ws_connection, producer: Awaitable, consumer: Awaitable, *args, **kwargs
```

Comment:
Typehint `ws_connection` and use a generic on `Awaitable` to denote the return types. Also why is `*args, **kwargs` added when never used?